### ISSUE TITLE
Feature: Upsert into table

### ISSUE THREAD

**Issue by trentkg** [DataTables/DataTables#981]
I'm using datatables and socketio for real time tables. It would be awesome if there was a way to upsert data into a datatable! I'm sure I can write my own upsert function but things are always better if they're built into the API.

**Comment by DataTables** [comment]
Yup - good feature request - I can see this being useful. For things like this I would normally write a plug-in API method and then if it gains enough interest / usage it would be integrated into the core.
